.github/workflows/ci.yaml

```diff
@@ -11,7 +11,7 @@ jobs:
         uses: actions/setup-java@v2
         with:
           distribution: 'adopt'
-          java-version: '17'
+          java-version: '21'
```

Comment:
I don't know yet what exactly requires java 21 in this PR, but we need to stick to java 17 (should be compatible with java 17), because it's used in Compose repository.